query-engine/connector-test-kit/src/test/scala/writes/relations/CompoundForeignKeysWithMixedRequiredness.scala

```diff
@@ -0,0 +1,69 @@
+package writes.relations
+
+import org.scalatest.{FlatSpec, Matchers}
+import util.ConnectorCapability.JoinRelationLinksCapability
+import util._
+
+class CompoundForeignKeysWithMixedRequiredness extends FlatSpec with Matchers with ApiSpecBase {
+  override def runOnlyForCapabilities = Set(JoinRelationLinksCapability)
+
+  "A One to Many relation with mixed requiredness" should "be writable and readable" in {
+    val testDataModels = {
+      val dm1 = """
+                  model Post {
+                      id       Int   @id @default(autoincrement())
+                      user_id  Int
+                      user_age Int?
+                      User     User? @relation(fields: [user_id, user_age], references: [id, age])
+
+                  }
+
+                  model User {
+                      id   Int    @id @default(autoincrement())
+                      age  Int
+                      Post Post[]
+
+                      @@unique([id, age], name: "user_unique")
+                  }"""
+
+      val dm2 = """
+                  model Post {
+                      id       Int   @id @default(autoincrement())
+                      user_id  Int
+                      user_age Int?
+                      User     User? @relation(fields: [user_id, user_age], references: [id, age])
+
+                  }
+
+                  model User {
+                      id   Int    @id @default(autoincrement())
+                      age  Int
+                      Post Post[]
+
+                      @@unique([id, age], name: "user_unique")
+                  }"""
+      TestDataModels(mongo = dm1, sql = dm2)
+    }
+
+    testDataModels.testV11 { project =>
+      //Setup user
+      server.query("mutation{createUser(data:{id: 1, age: 1}){id, age, Post{id}}}", project).toString() should be(
+        """{"data":{"createUser":{"id":1,"age":1,"Post":[]}}}""")
+
+      //Input not allowed violation
```

Comment:
This one we might want to handle via changing the CreateInputTypes according to Dom. I don't think this needs to happen at the same time necessarily. Just changing the introspection behavior / validation to switch mixed required ness relations to optional is already an improvement I think.

Reply:
No, doesn't need to happen at the same time, it's a separate feature. As it is right now is okay, the QE team will take it from there as a new feature.